sunpy/util/util.py

```diff
@@ -195,3 +197,32 @@ def hash_file(path):
             sha256.update(data)
 
     return sha256.hexdigest()
+
+
+def get_rectangle_coordinates(bottom_left, *, top_right = None, width: u.deg = None, height: u.deg = None):
+
+    if (not isinstance(bottom_left, (SkyCoord, BaseCoordinateFrame)) and 
+        not isinstance(top_right, (None, SkyCoord, BaseCoordinateFrame))):
+        raise TypeError("bottom_left and top_right must be of type SkyCoord or BaseCoordinateFrame")
+
+    elif bottom_left.shape == (2,):
+        top_right = bottom_left[1]
+        bottom_left = bottom_left[0]
+
+    elif top_right is not None:
+        width = top_right.spherical.lon - bottom_left.spherical.lon   # Getting the difference in Longitudes.
```

Comment:
The comments here are probably not needed, it's reasonably obvious that's what happening from the code.